core/src/test/scala/unit/kafka/integration/UncleanLeaderElectionTest.scala

```diff
@@ -112,7 +112,6 @@ class UncleanLeaderElectionTest extends ZooKeeperTestHarness {
   }
 
   @Test
-  @Ignore // Should be re-enabled after KAFKA-3096 is fixed
```

Comment:
Do you know how the test fails w/o this PR?

Reply:
In this testcase, we are creating a topic with single partition and RF=2. We stop the follower, send messages, shutdown leader and then restart follower. Now we are checking the leader to become -1.

Previously, we are not transitioning offline replicas during startup, so leader value contain old leader value. In this PR, we are making new leader to -1 during offline replica transition.